injected/src/features/message-bridge.js

```diff
@@ -0,0 +1,168 @@
+     */
+    installProxyFor(featureName, config) {
+        if (this.proxies.has(featureName)) return;
+        const allowed = this.getFeatureSetting('allowed');
+        if (!allowed.includes(featureName)) return;
+
+        const ctx = { ...this.messaging.messagingContext, featureName };
+        const messaging = new Messaging(ctx, config);
+        this.proxies.set(featureName, messaging);
+        console.log('[isolated] did install proxy for ', featureName);
+    }
+
+    /**
+     * @param {ProxyRequest} request
+     * @param {(s: string) => string} appendToken
+     */
+    proxyRequest(request, appendToken) {
+        const { id, featureName, method, params } = request;
+
+        const proxy = this.proxies.get(featureName);
+        if (!proxy) return console.warn('[isolated] proxy was not installed');
+
+        const responseName = appendToken(ProxyResponse.NAME + '-' + id);
+        console.log('[isolated] will proxy', request);
+
+        proxy
+            ?.request(method, params)
+            // eslint-disable-next-line promise/prefer-await-to-then
+            .then((response) => {
+                const responseEvent = new ProxyResponse({
+                    method,
+                    featureName,
+                    result: response,
+                    id,
+                });
+                const evt = new CustomEvent(responseName, { detail: responseEvent });
+                console.log('[isolated] will dispatch', evt);
+                window.dispatchEvent(evt);
+            })
+            // eslint-disable-next-line promise/prefer-await-to-then
+            .catch((e) => {
+                const errorResponseEvent = new ProxyResponse({
+                    method,
+                    featureName,
+                    error: { message: e.message },
+                    id,
+                });
+                const evt = new CustomEvent(responseName, { detail: errorResponseEvent });
+                window.dispatchEvent(evt);
+            });
+    }
+
+    /**
+     * @param {SubscriptionRequest} subscription
+     * @param {(s: string) => string} appendToken
+     */
+    proxySubscription(subscription, appendToken) {
+        const { id, featureName, subscriptionName } = subscription;
+        const proxy = this.proxies.get(subscription.featureName);
+        if (!proxy) return console.warn('[isolated] proxy was not installed');
+
+        const responseName = appendToken(SubscriptionResponse.NAME + '-' + id);
+
+        console.log('[isolated] will setup subscription', subscription);
+        this.subscriptions.set(
+            id,
+            proxy.subscribe(subscriptionName, (data) => {
+                const responseEvent = new SubscriptionResponse({
+                    subscriptionName,
+                    featureName,
+                    params: data,
+                    id,
+                });
+                const evt = new CustomEvent(responseName, { detail: responseEvent });
```

Comment:
Does this not need the wrapped name also?

Reply:
Maybe make a central dispatch method rather than calling createEvent and dispatch in many places?

Reply:
agreed that a central dispatch method sounds nice.